opslevel/resource_opslevel_team.go

```diff
@@ -1,311 +1,322 @@
-// 	id := d.Id()
-// 	err := client.DeleteTeam(id)
-// 	if err != nil {
-// 		return err
-// 	}
-// 	d.SetId("")
-// 	return nil
-// }
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+var _ resource.ResourceWithConfigure = &TeamResource{}
+
+var _ resource.ResourceWithImportState = &TeamResource{}
+
+type TeamResource struct {
+	CommonResourceClient
+}
+
+func NewTeamResource() resource.Resource {
+	return &TeamResource{}
+}
+
+// TeamResourceModel describes the Team managed resource.
+type TeamResourceModel struct {
+	Aliases          types.List   `tfsdk:"aliases"`
+	Id               types.String `tfsdk:"id"`
+	LastUpdated      types.String `tfsdk:"last_updated"`
+	Member           []TeamMember `tfsdk:"member"`
+	Name             types.String `tfsdk:"name"`
+	Parent           types.String `tfsdk:"parent"`
+	Responsibilities types.String `tfsdk:"responsibilities"`
+}
+
+type TeamMember struct {
+	Email types.String `tfsdk:"email"`
+	Role  types.String `tfsdk:"role"`
+}
+
+func convertTeamMember(teamMember opslevel.TeamMembership) TeamMember {
+	return TeamMember{
+		Email: RequiredStringValue(teamMember.User.Email),
+		Role:  RequiredStringValue(teamMember.Role),
+	}
+}
+
+func NewTeamResourceModel(ctx context.Context, team opslevel.Team, parent string) (TeamResourceModel, diag.Diagnostics) {
+	aliases, diags := OptionalStringListValue(ctx, team.ManagedAliases)
+	if diags != nil && diags.HasError() {
+		return TeamResourceModel{}, diags
+	}
+	if len(aliases.Elements()) == 0 {
+		aliases = types.ListNull(types.StringType)
+	}
+	teamMembers := make([]TeamMember, 0)
+	if team.Memberships != nil {
+		for _, mem := range team.Memberships.Nodes {
+			teamMembers = append(teamMembers, convertTeamMember(mem))
+		}
+	}
+	teamResourceModel := TeamResourceModel{
+		Aliases: aliases,
+		Id:      types.StringValue(string(team.Id)),
+		Member:  teamMembers,
+		Name:    types.StringValue(team.Name),
+	}
+	if team.ParentTeam.Alias != "" && team.ParentTeam.Id != "" {
+		// use an ID or alias for this field based on what is currently in the state
+		if opslevel.IsID(parent) {
+			teamResourceModel.Parent = types.StringValue(string(team.ParentTeam.Id))
+		} else {
+			// TODO: there is a case where the user is using an alias from the parent team that is not the default alias
+			teamResourceModel.Parent = types.StringValue(team.ParentTeam.Alias)
+		}
+	}
```

Comment:
There seem to be checks on two different parent sources in this if statement. Would something like this work?
```go
	if opslevel.IsID(string(team.ParentTeam.Id)) {
		teamResourceModel.Parent = types.StringValue(string(team.ParentTeam.Id))
	} else {
		teamResourceModel.Parent = OptionalStringValue(team.ParentTeam.Alias)
	}
```
The `if team.ParentTeam.Alias != "" && team.ParentTeam.Id != "" {` part could be dropped with this too

Reply:
@davidbloss Added your suggestion but the `if` needs to use the `parent` arg:

```
	// if parent is set, use an ID or alias for this field based on what is currently in the state
	if opslevel.IsID(parent) {
		teamResourceModel.Parent = types.StringValue(string(team.ParentTeam.Id))
	} else {
		// TODO: error thrown if config has alias from the parent team that is not the default alias
		teamResourceModel.Parent = OptionalStringValue(team.ParentTeam.Alias)
	}

```

Confirmed works by tophatting